integration/e2e/api_test.go

```diff
@@ -56,111 +57,220 @@ func TestSearchTagValuesV2(t *testing.T) {
 			tagName: spanX,
-			expected: searchTagValuesResponse{
+			expected: searchTagValuesV2Response{
 				TagValues: []TagValue{{Type: "string", Value: firstBatch.spanAttVal}},
 			},
 		},
 		{
 			name:    "second batch only - resource attribute",
 			query:   fmt.Sprintf(`{ %s="%s" }`, resourceX, secondBatch.resourceAttVal),
 			tagName: spanX,
-			expected: searchTagValuesResponse{
+			expected: searchTagValuesV2Response{
 				TagValues: []TagValue{{Type: "string", Value: secondBatch.spanAttVal}},
 			},
 		},
 		{
 			name:     "too restrictive query",
 			query:    fmt.Sprintf(`{ %s="%s" && resource.y="%s" }`, resourceX, firstBatch.resourceAttVal, secondBatch.resourceAttVal),
 			tagName:  spanX,
-			expected: searchTagValuesResponse{},
+			expected: searchTagValuesV2Response{},
 		},
 		// Unscoped not supported, unfiltered results.
 		{
 			name:    "unscoped span attribute",
 			query:   fmt.Sprintf(`{ .x="%s" }`, firstBatch.spanAttVal),
 			tagName: spanX,
-			expected: searchTagValuesResponse{
+			expected: searchTagValuesV2Response{
 				TagValues: []TagValue{{Type: "string", Value: firstBatch.spanAttVal}, {Type: "string", Value: secondBatch.spanAttVal}},
 			},
 		},
 		{
 			name:    "unscoped resource attribute",
 			query:   fmt.Sprintf(`{ .xx="%s" }`, firstBatch.spanAttVal),
 			tagName: resourceX,
-			expected: searchTagValuesResponse{
+			expected: searchTagValuesV2Response{
 				TagValues: []TagValue{{Type: "string", Value: firstBatch.resourceAttVal}, {Type: "string", Value: secondBatch.resourceAttVal}},
 			},
 		},
 		{
 			name:    "first batch - name and resource attribute",
 			query:   fmt.Sprintf(`{ name="%s" }`, firstBatch.name),
 			tagName: "resource.service.name",
-			expected: searchTagValuesResponse{
+			expected: searchTagValuesV2Response{
 				TagValues: []TagValue{{Type: "string", Value: "my-service"}},
 			},
 		},
 		{
 			name:    "both batches - name and resource attribute",
 			query:   `{ resource.service.name="my-service"}`,
 			tagName: "name",
-			expected: searchTagValuesResponse{
+			expected: searchTagValuesV2Response{
 				TagValues: []TagValue{{Type: "string", Value: secondBatch.name}, {Type: "string", Value: firstBatch.name}},
 			},
 		},
 		{
 			name:    "only resource attributes",
 			query:   fmt.Sprintf(`{ %s="%s" }`, resourceX, firstBatch.resourceAttVal),
 			tagName: "resource.service.name",
-			expected: searchTagValuesResponse{
+			expected: searchTagValuesV2Response{
 				TagValues: []TagValue{{Type: "string", Value: "my-service"}},
 			},
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			callSearchTagValuesAndAssert(t, tempo, tc.tagName, tc.query, tc.expected)
+			callSearchTagValuesV2AndAssert(t, tempo, tc.tagName, tc.query, tc.expected, 0, 0)
 		})
 	}
+
+	// Wait to block flushed to backend, 20 seconds is the complete_block_timeout configuration on all in one, we add
+	// 1s for security.
```

Comment:
```suggestion
	// Wait to block flushed to backend, 20 seconds is the complete_block_timeout configuration on all in one, we add
	// 10s for security.
```

10s might be too much. I'd go for 1-2s and assert that the block has been flushed to the backend and cleared from the ingester with metrics (`tempo_ingester_blocks_flushed_total`, `tempodb_blocklist_length`, `tempo_ingester_blocks_cleared_total`)

Reply:
See a similar example in https://github.com/grafana/tempo/blob/main/integration/e2e/e2e_test.go#L118-L121